### ISSUE TITLE
Quarterly donations

### ISSUE THREAD

**Issue by woj-tek** [liberapay/liberapay.com#1581]
Currently it's possible to donate weekly/monthly/yearly. IMHO monthly would be OK, but the commission on transaction/transfer make it a so-so option. On the other hand, yearly seems spread to far.

Would it be possible to have third option: quarterly to strike better balance?

**Comment by Changaco** [comment]
The payment schedule isn't tied to the donation period. You can pay a weekly donation quarterly, and you can pay a yearly donation monthly.

**Comment by woj-tek** [comment]
> You can pay a weekly donation quarterly, and you can pay a yearly donation monthly.

I'm sorry, but this is rather confusing. Is there any ELI5 explanation? Aren't donations automatic?

**Comment by mattbk** [comment]
There are 13 weeks in a quarter. If you want to donate $20 per quarter, set your donation as $20/13, or $1.54 per week. You'll pay $20 up front, the other person will receive $20 up front, and you'll be reminded to renew your donation in 13 weeks.

**Comment by woj-tek** [comment]
OK, still not getting it. I can select: per week, per month (and considering quarters as 3 months would be easier in this case), per year. But if I decide to donate $12 quarterly to some project that means $4 monthly. If I select $4 monthly then transaction fees would eat up significant chunk of the donation. If I select yearly then: I pay more at once, and the project receives more at once (and I do believe that constant stream would be better). The idea to do it quarterly was mainly to balance the fees - if not for them I'd end up doing it monthly (but then there is this scary warning "fees are huuuuge, don't do it".

Ideally I should be able to select a bunch of project with monthly donation, I'd be charged *once* with the sum of all donations for the month (and it could even have added fee so all librepay receives could be easily distributed to projects - and the fee is usually 3% with minimum of $0,5 and that minimum makes it completely pointless to make small, distinct donations, but with ~$25 monthly the fee would be still under $1 AFAIK).

From my observation, patron is doing just that and it's kinda make sense.

**Comment by Changaco** [comment]
A $4 per month donation isn't necessarily billed as $4 monthly, just like a $0.01 weekly donation is obviously not billed as $0.01 every week.

When a donor chooses a $4 monthly donation, our software suggests 3 different payment schedules: $4 per month (high fee), $12 per quarter (moderate fee), and $48 per year (low fee).

![Screenshot of a payment form for a $4 monthly donation](https://user-images.githubusercontent.com/1581590/65223727-1ad22100-dac2-11e9-8425-9f0043c32f15.png)

**Comment by woj-tek** [comment]
Problem being - that screen (with quarterly donation) is pure chance. For example one may see something like this:
<img width="811" alt="librepay" src="https://user-images.githubusercontent.com/724413/65253731-853e8d80-dad1-11e9-9f86-c1ca9957ba9c.png">
(I selected the smallest possible donation and this was an end result).

Bottomline is - this is automatic and I have no control when I will be charged and when the project will receive the funding. To boot, it's quite convoluted.

What I'm looking for is something simple:
* group all donations so there will be only one charge (with relatively small fee)
* do it monthly or quarterly (depending on amount of money that would be donated vs fees).

IMHO suggesting x-yearly donations just to avoid fees seems weird.

**Comment by mattbk** [comment]
@woj-tek, unfortunately, it's still not possible to bundle donations to different recipients under one transaction to save on fees (this is not by choice, but a function of the payment processors being used).

> But if I decide to donate $12 quarterly to some project that means $4 monthly.

See @Changaco's screenshot above for an example of how a $4 monthly donation can be donated quarterly. You're not actually moving $4 per month, you're paying $12(+fee) up front, the recipient gets $12 up front, and you're reminded to pay again in three months.

>  just like a $0.01 weekly donation is obviously not billed as $0.01 every week.

Except that this is not obvious, because this was how it worked under the wallet system, and how it's expressed in the UI. I think this question is going to keep coming up (#1205, #483). It seems like the UX could be changed to something like:

- How much to give bob: $12
- When would you like to be reminded to donate again: 3 months*
- This averages out to: $4 per month, $1 per week, and if you renew, $48 per year*

Then on bob's page, it could still report how much received per week based on the math.



*my date math is poor.

**Comment by woj-tek** [comment]
> unfortunately, it's still not possible to bundle donations to different recipients under one transaction to save on fees (this is not by choice, but a function of the payment processors being used).

I understand the limitation and I saw a ticket with information that this should be address in the future (it's already possible with EU groups)

I agree that current UI is rather unfortunate and what you @mattbk suggested above makes more sense. Fingers crossed that with grouping of payments it will be possible to have montly donation and monthly charge what would not be eaten-up by fees.

Btw. have you considered moving fees to the donor (and displaying them)? That could help making more informed decision about the amount that would be lost (and that I would be willing to lose)...

**Comment by Changaco** [comment]
@woj-tek

> Problem being - that screen (with quarterly donation) is pure chance.

It's not “pure chance”, the software is designed to suggest quarterly payments when they make sense.

> For example one may see something like this:

That's the worst case. All other donation amounts result in payment suggestions that are saner, which is why I've been considering doubling the “Symbolic” donation amount (#975).

> Bottomline is - this is automatic and I have no control when I will be charged and when the project will receive the funding.

You have much more control over when you're charged than if you were using a platform that only supports charging you automatically every month (e.g. Patreon).

@mattbk

> Except that this is not obvious, because this was how it worked under the wallet system

Donors have always been charged upfront, the difference is that the funds used to be sent to the recipients over time whereas now they're transferred immediately.

I meant “obviously” in the context of the current discussion, I know that it's not obvious in general.

> It seems like the UX could be changed to something like:

I've opened a new issue to discuss the evolution of the donation form: #1584.

**Comment by Changaco** [comment]
> Btw. have you considered moving fees to the donor (and displaying them)?

We're currently unable to compute the fee, we only know how much it is after the payment has been processed.
